### ISSUE TITLE
Channel name removed from bold text

### ISSUE THREAD

**Issue by paulchen** [RocketChat/Rocket.Chat.ReactNative#5305]
### Description:

When a channel name used inside bold text, it will not be shown.

### Environment Information:

- Rocket.Chat Server Version: 6.4.4
- Rocket.Chat App Version: 4.42.0.50842
- Device Name: ELE-L29 (Huawei P30)
- OS Version: EMUI 12.0.0 (Android 11)

### Steps to reproduce:

1. Write the message `* #channel *`.

### Expected behavior:

The channel name is shown in the same way as in the browser:

![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/404840/31b56809-3bbd-4e03-8a9e-453bdfd8b26d)

### Actual behavior:

The channel name is not shown:

![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/404840/7a8b58cf-e2ff-4d0f-b879-6bc07d47188f)

However, it is shown in the channel list:

![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/404840/298b75d5-be87-4dba-9c21-f72b1ad0fbb4)

And also when quoting the message:

![image](https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/404840/f1fc5df7-8337-4763-9b28-8cc42e343fc4)

(in the quote, the text is not formatted in bold face, but instead as an enumeration)

### Additional context:

This also applies for italics and strikethrough. The behaviour does not depend on whether the channel actually exists.

**Comment by brf153** [comment]
@paulchen please assign this to me.

**Comment by paulchen** [comment]
> @paulchen please assign this to me.

Sorry, I'm afraid I can't. I'm not a member of the Rocket.Chat team.

Besides that, to my knowledge the Rocket.Chat team does not assign issues to externals. However, feel free to fix this issue and submit a pull request for it.

**Comment by brf153** [comment]
oki

**Comment by brf153** [comment]
@diegolmello , I realized that to make this change, we need to modify the definitions.d.ts file located in the node_modules directory. We should include ChannelMention in the Markup type definition within that file. Then, in the Bold.tsx file, we need to add a case for MENTION_CHANNEL where we can use the Plain component, passing the value as <Plain value={#${block.value.value}} />. I have made these changes and will be submitting a pull request (PR) for the same. I just need to confirm where I can access and edit the definitions.d.ts file.
